pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -225,29 +246,46 @@ def save_model(self, filepath, overwrite):
 
     def on_epoch_end(self, epoch, logs=None):
         logs = logs or {}
-        self.epochs_since_last_save += 1
-        if self.epochs_since_last_save >= self.period:
-            self.epochs_since_last_save = 0
+        self.epochs_since_last_check += 1
+        if self.epochs_since_last_check >= self.period:
+            self.epochs_since_last_check = 0
             filepath = '{}/{}_ckpt_epoch_{}.ckpt'.format(self.filepath, self.prefix, epoch + 1)
-            if self.save_best_only:
+            if self.save_top_k:
                 current = logs.get(self.monitor)
                 if current is None:
                     print('Can save best model only with %s available,'
                           ' skipping.' % (self.monitor), RuntimeWarning)
                 else:
-                    if self.monitor_op(current, self.best):
+                    if ((len(self.bestk.keys()) < self.save_top_k) or
```

Comment:
pls cut-out this functionality to separate (private) method, for luck of reading